### ISSUE TITLE
`_get_analytical_jacobian_forward_ad` cannot compute gradient for `nn.MaxPool`

### ISSUE THREAD

**Issue by upupyy** [pytorch/pytorch#78342]
### 🐛 Describe the bug

`_get_analytical_jacobian_forward_ad` cannot compute gradient for `nn.MaxPool`.

```python
# add in the file torch/autograd/gradcheck.py
import torch
from torch.autograd import forward_ad

def get_fn():
    kernel_size = 2
    stride = 2
    return_indices = False
    arg_class = torch.nn.MaxPool1d(kernel_size, stride=stride, return_indices=return_indices)
    def fn(input):
        fn_res = arg_class(input)
        return fn_res
    return fn
fn = get_fn()
input_tensor = torch.rand([1, 1, 8], dtype=torch.float32)

input = input_tensor.clone().requires_grad_()
output = fn(input)
res = _get_analytical_jacobian_forward_ad(fn, (input,), (output,))
print(res)
# ((tensor([[0., 0., 0., 0.],
#         [0., 0., 0., 0.],
#         [0., 0., 0., 0.],
#         [0., 0., 0., 0.],
#         [0., 0., 0., 0.],
#         [0., 0., 0., 0.],
#         [0., 0., 0., 0.],
#         [0., 0., 0., 0.]]),),)


input = input_tensor.clone().requires_grad_()
tangents = []
for i in range(8):
    tensor = torch.zeros([1, 1, 8])
    tensor[0][0][i] = 1
    tangents.append(tensor)
with forward_ad.dual_level():
    for tangent in tangents:
        dual_input = forward_ad.make_dual(input, tangent)
        dual_output = fn(dual_input)
        column = forward_ad.unpack_dual(dual_output).tangent
        print(column)
# tensor([[[0., 0., 0., 0.]]])
# tensor([[[1., 0., 0., 0.]]])
# tensor([[[0., 1., 0., 0.]]])
# tensor([[[0., 0., 0., 0.]]])
# tensor([[[0., 0., 1., 0.]]])
# tensor([[[0., 0., 0., 0.]]])
# tensor([[[0., 0., 0., 1.]]])
# tensor([[[0., 0., 0., 0.]]])
```

The forward-mode gradient computed by `_get_analytical_jacobian_forward_ad` is wrong.

When analyzing the root cause, I am confused by the comment of `_get_analytical_jacobian_forward_ad` that "The computed value is expected to be wrong if the function captures the inputs by side effect instead of using the passed ones (many torch.nn tests do this)."

Does it mean this result is expected?

### Versions

pytorch: 1.11.0

**Comment by upupyy** [comment]
It seems that this has been fixed in the nightly version.
